File: epargne/Views/CompteEpargneView.cs
using System;
using System.Collections.Generic;
using Epargne.DTO;
using Epargne.Models;

namespace Epargne.Views
{
    public class CompteEpargneView
    {
        public int IdCompte { get; set; }
        public int IdClient { get; set; }

        public decimal CapitalEpargne { get; set; }
        public string Libelle { get; set; }
        public DateOnly DateOuverture { get; set; }
        public decimal TauxInteret { get; set; }
        public decimal Retrait { get; set; }

        public decimal Solde { get; set; } = 0.0m;

        public List<TransactionEpargneDto> Transactions { get; set; }
    }
}

File: server-ejb/src/main/java/com/example/views/CompteEpargneView.java
package com.example.views;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

import com.example.dto.TransactionEpargneDto;

import lombok.Data;

@Data
public class CompteEpargneView implements Serializable {

    private int idCompte;
    private int idClient;

    private BigDecimal capitalEpargne;
    private String libelle;
    private LocalDate dateOuverture;
    private BigDecimal tauxInteret;
    private BigDecimal retrait;
    private BigDecimal solde = BigDecimal.ZERO;

    private List<TransactionEpargneDto> transactions;
}
